Two frontends, both alike in dignity, in fair VHS where we lay our scene.

coryalder opened the pull request:
In an effort to land the feat/writing-a-lot-of-code-for-v2.0-and-all-I-got-was-this-lousy-branch-name branch, with it's majestic react frontend lovelyness.

My plan was (and still is) to deploy both frontends - 

- `https://membership.vanhack.ca/` the new react version
- `https://membership.vanhack.ca/v1/`  the old angular version

todo:

- [ ] react admin can't edit a member's privs, broken
- [ ] react admin can't create new privs, missing
- [ ] react admin can't delete privs, missing
- [ ] react user can't search users on grant privileges page, broken
- [x] react user can't grant privs, broken
- [x] add an "old interface" button somewhere, so you can escape if needed
- [ ] deploy!

TyIsI committed:
build(backend): fix missing php dependencies

TyIsI committed:
build(backend): improved react nginx configuration

TyIsI committed:
feat(backend): improved logging in AWSSESClient

TyIsI committed:
refactor(frontend): optimize schema usage in pages

TyIsI committed:
style(frontend): remove border from buttons

TyIsI committed:
refactor(backend): AWSSESClient phpdoc and renamed recipients param

TyIsI committed:
feat(backend): EmailAdapter

TyIsI committed:
refactor(backend): moved Email logic from EmailServiceHandler2 to EmailAdapter

TyIsI committed:
fix(backend): inadvertent new statement in Loggington moved to proto construct

TyIsI committed:
refactor(backend): EmailAdapter param order with subject default null

TyIsI committed:
refactor(backend): UserServiceHandler and PaymentProcessor to use EmailAdapter

TyIsI committed:
fix(backend): updated user not being returned on user creation

TyIsI committed:
fix(frontend): issues with creating and editing new users due to nullish values

TyIsI committed:
feat(backend): add grants permission to GetUserPrivileges service method

TyIsI committed:
refactor(frontend): renamed/optimized validators and split guards output files

TyIsI committed:
style(frontend): removed redundant spacious statements from Row calls

TyIsI committed:
feat(frontend): initial UserGranting page implementation

TyIsI committed:
feat(frontend): useCountUsers hook

TyIsI committed:
feat(frontend): useGetAllPrivileges hook

TyIsI committed:
feat(frontend): useGetGrantUserPrivileges hook

TyIsI committed:
feat(frontend): useGetUser hook

TyIsI committed:
feat(frontend): useGetUserPrivileges hook

TyIsI committed:
feat(frontend): useListUsers hook

TyIsI committed:
fix(frontend): change membership property type in User db model

TyIsI committed:
fix(frontend): possibly null membership field value in CurrentUserPrivilegesCard

TyIsI committed:
feat(frontend): wrap actions in Cols in BasePase

TyIsI committed:
feat(frontend): PrivilegeMutation type

TyIsI committed:
style(frontend): improved pull-left and pull-right handling

TyIsI committed:
refactor(frontend): separated TablePageSchema search key into TablePageSearchSchema

TyIsI committed:
refactor(frontend): changed onUpdate mutation param to shared PrivilegeMutation type

TyIsI committed:
clean(frontend): removed unnecessary comment from fetcher

TyIsI committed:
dev(frontend): expand eslint max-params limit to 8

TyIsI committed:
fix(frontend): missing dependency and await in UserGrantingItem

TyIsI committed:
chore(frontend): formatting

TyIsI committed:
dev(frontend): add src/types/nomos.d.ts to prettierignore

TyIsI committed:
fix(frontend): apikeys stuck on loading without existing keys

TyIsI committed:
refactor: regenerate service provider implementations/types

TyIsI committed:
feat: improved filters parameter typing between backend and frontend

TyIsI committed:
fix(frontend): filters parameter types in UserService2 hooks

TyIsI committed:
chore(frontend): added missing mutateListUsers prop to UserGrantingItem test

TyIsI committed:
chore(frontend): add mutateListUsers to UserGrantingItem prop types

TyIsI committed:
refactor(frontend): use GetUserGrantablePrivileges in UserGrantingItem

TyIsI committed:
feat: fixed type issues for UserGranting

TyIsI committed:
feat(backend): HttpStatusCodes enums

TyIsI committed:
feat(backend): HttpException

TyIsI committed:
refactor(backend): change InvalidCredentials and InvalidRequestException to extend HttpException

TyIsI committed:
refactor: revamp and COMMENTS EVERYWHERE

TyIsI committed:
dev: improved phpstan config

TyIsI committed:
chore: composer lockfile update

TyIsI committed:
fix(backend): workarounds for null params before non-null params

TyIsI committed:
feat(backend): add Migrator overrideable options support

TyIsI committed:
chore: bump packageManager

TyIsI committed:
refactor(backend): split Domain/Schema classes and interfaces to separate files

TyIsI committed:
chore(backend): reformat/reorder Migrator.php

TyIsI committed:
clean(backend): remove unused imports and comments

TyIsI committed:
refactor(backend): refactored and documented/commented tests

TyIsI committed:
chore(backend): formatting and lots of comments

TyIsI committed:
chore(backend): fixed Cloneable inline documentation

TyIsI committed:
chore(backend): fixed Migrator inline typing

TyIsI committed:
dev(backend): enabled no_unused_imports php-cs-fixer setting

TyIsI committed:
refactor(backend): typing, array conversion, and type comments

TyIsI committed:
dev: improved phpstan config

TyIsI committed:
dev: improved php-cs-fixer config

TyIsI committed:
dev: editorconfig formatting

TyIsI committed:
dev: fixed config issue in phpstan config

TyIsI committed:
chore(backend): add return phpdoc statement in WebHookServiceHandler2.php

TyIsI committed:
dev(frontend): improved eslint/ts config for vite.config.ts support

TyIsI committed:
dev(frontend): improved vite config host overrides

TyIsI committed:
feat(backend): improved HttpExceptions

TyIsI committed:
feat(backend): improved MemberCardService2 handling (method rename+improved exception)

TyIsI committed:
feat(frontend): improved exception handling for failed fetches

TyIsI committed:
feat(frontend): embedded support for BasePage component

TyIsI committed:
feat(frontend): improved TablePage with embedded support, default filters and enhanced visual handling

TyIsI committed:
feat(frontend): MemberCardSerialNumber validator

TyIsI committed:
style(frontend): tight-vertical utility class

TyIsI committed:
style(frontend): improved button content scaling regular vs small

TyIsI committed:
feat(frontend): MemberCard admin page basic working version

TyIsI committed:
dev(frontend): fixed eslint ts node types handling

TyIsI committed:
chore(frontend): formatting for AdminMemberCards pages/components

TyIsI committed:
refactor(frontend): pinned HTTPException info field to string

TyIsI committed:
dev: fixed husky setup detection issue

TyIsI committed:
repo: package/pnpm bump

TyIsI committed:
chore: formatting

TyIsI committed:
fix(frontend): HTTPException data/info fixes

TyIsI committed:
feat(frontend): improved pagination and pagination layout

TyIsI committed:
dev(frontend): tweaking eslint config (removed fafo definitions)

TyIsI committed:
fix(backend): incorrect type check in PHP2TS convertDocComment method

TyIsI committed:
feat(backend): add updateTemplate service method to v2 Email service

TyIsI committed:
chore(frontend): synced service implementations and interfaces from backend

TyIsI committed:
feat(frontend): make Popover content wrap

TyIsI committed:
feat(frontend): automatically make ConditionalTableCell content Popover based on string length

TyIsI committed:
feat(frontend): remove FormControl height enforcement over minimum height

TyIsI committed:
feat(frontend): refactor table row/cell width scaling and allow cell width overrides by cols prop

TyIsI committed:
feat(frontend): zNormativeString validator implements string with at least 3 characters

TyIsI committed:
feat(frontend): use zNormativeString for EmailTemplate datatype

TyIsI committed:
refactor(frontend): AdminAccessLogsItem to use TablePageRow

TyIsI committed:
feat(frontend): initial useGetTemplate implementation

TyIsI committed:
feat(frontend): AdminEmailTemplate pages implementation

TyIsI committed:
chore(backend): formatting

TyIsI committed:
feat(frontend): finished AdminMemberCard pages implementation

TyIsI committed:
chore: nginx vhost configurations formatting

TyIsI committed:
dev(frontend): enable eslint jsx-a11y support for FormControl as a label target

TyIsI committed:
chore: add web directory to prettier ignore

TyIsI committed:
feat: split OAuth service methods from AuthService2 into OAuthService2 (and generated frontend provider implementations)

TyIsI committed:
tools(frontend): integrated generation of template provider hooks into provider implementations

TyIsI committed:
feat(frontend): regenerated provider hooks

TyIsI committed:
feat(frontend): added OAuthService2 as valid table page endpoint type

TyIsI committed:
feat(frontend): tweaked zAppClient field length requirements

TyIsI committed:
feat(backend): OAuthService2 service endpoint

TyIsI committed:
fix(frontend): rename useGetAllMemberships to useGetAll hook

TyIsI committed:
feat: OAuthService2 UpdateClient and UpdateClientExpiry service methods

TyIsI committed:
chore(frontend): TODO comment for Popover component

TyIsI committed:
feat(frontend): Button circle prop support

TyIsI committed:
feat(frontend): TableActionCell className override support

TyIsI committed:
feat(frontend): pass fieldStates to component in TablePage

TyIsI committed:
refactor(frontend): TablePage Item components to use fieldStates and TablePageItemComponent interface

TyIsI committed:
tools(frontend): optimized generate-provider-implementations script

TyIsI committed:
fix(frontend): AdminUsersItem fields

TyIsI committed:
feat: OAuthService2 GetClientDetails service method (and frontend implementations updates)

TyIsI committed:
feat(frontend): LinkButton circle prop support

TyIsI committed:
feat(frontend): initial OAuth management page implementation

TyIsI committed:
feat: change OAuthService2 GetClientDetails service method argument name

TyIsI committed:
tools(frontend): improve generate-provider-implementations parallelization

TyIsI committed:
refactor(frontend): ConditionalTableCell to TableDataCell and improved Item component consistency

TyIsI committed:
feat(frontend): ItemDeleteModal component

TyIsI committed:
feat(frontend): ItemDeleteModal overrideable title

TyIsI committed:
refactor(frontend): components to use ItemDeleteModal

TyIsI committed:
feat(frontend): implemented useGetPRivilege

TyIsI committed:
feat(frontend): implemented and refactored to AdminPrivilegesEdit page

TyIsI committed:
fix(frontend): ItemDeleteModal test props

TyIsI committed:
refactor(frontend): renamed WebHooksComponent to WebHooksItem

TyIsI committed:
refactor(frontend): make /admin/oauth use lazy loading

TyIsI committed:
fix(frontend): remove dynamic toggle from AdminPrivilegesItem

TyIsI committed:
refactor(frontend): enforce Lazy component name suffix

TyIsI committed:
refactor(frontend): renamed page components to naming convention

TyIsI committed:
feat(frontend): change user access history url

TyIsI committed:
feat(frontend): make user dashboard lazy loading

TyIsI committed:
refactor(frontend): renamed EmailTemplateCreate to EmailTemplateNew

TyIsI committed:
chore: format all

TyIsI committed:
refactor(frontend): change component import paths

TyIsI committed:
feat(frontend): EnabledCheckMark component

TyIsI committed:
refactor(frontend): use EnabledCheckMark for boolean fields

TyIsI committed:
build: removed nomos- prefixes from compose configurations

TyIsI committed:
dev: added explicit prettier support for docker/nomos.env.template

TyIsI committed:
chore: corepack up

TyIsI committed:
build: refactor of compose configurations

TyIsI committed:
feat(frontend): empty string validator

TyIsI committed:
feat(frontend): ConfigProvider

TyIsI committed:
feat(frontend): implement ConfigProvider

TyIsI committed:
refactor(frontend): TablePageRow scaling

TyIsI committed:
feat(frontend): ensure default config.json file

TyIsI committed:
build(frontend): optimized build and scripts

TyIsI committed:
build(frontend): optimize build integration and scripts

TyIsI committed:
build: fix compose frontend-web build target

TyIsI committed:
feat(backend): stronger HttpException constructor code argument typing

TyIsI committed:
feat(backend): use HttpStatusCodes enum to set 200 http ok response

TyIsI committed:
chore(backend): formatting for inadvertently ignored vhs/web files

TyIsI committed:
dev: fix prettier ignore issue with inadvertently ignored vhs/web files

TyIsI committed:
feat(backend): improved InvalidRequestException handling with default Client_Error_Method_Not_Allowed code

TyIsI committed:
fix(backend): HttpJsonServiceHandlerModule would not correctly handle missing module exception

TyIsI committed:
fix(backend): HTTPServer invalid response code type exception

TyIsI committed:
docs(frontend): various TODO comments

TyIsI committed:
chore: formatting

TyIsI committed:
build: refactored docker-compose configurations for improved separation of concern

TyIsI committed:
dev: added docker-compose conf support to prettier config

TyIsI committed:
build: traefik http only configs

TyIsI committed:
build: compose backend local files configs

TyIsI committed:
build: renamed override.yml to override.sample.yml

TyIsI committed:
chore: diff-report tool and output

TyIsI committed:
chore: updated diff report

TyIsI committed:
fix(webhooker): undo the borkedness due to dependency upgrades

TyIsI committed:
deps(webhooker): various updates and dev dependencies

TyIsI committed:
feat(webhooker): updates

TyIsI committed:
chore: update pnpm-lock file

TyIsI committed:
refactor: move php and web code-bases to dedicated packages

TyIsI committed:
build: use DCT to generate Dockerfile

TyIsI committed:
build: removed test cache_from/to settings from backend-php compose build config

TyIsI committed:
build: fixed migrations directory path in backend-php compose config

TyIsI committed:
build: fixed label issues in frontend-react compose config

TyIsI committed:
build: missing conf files

TyIsI committed:
build(frontend-web): ensure installing web components

TyIsI committed:
fix(frontend-react): DoorAccess content

TyIsI committed:
feat(frontend-react): add valid field to user object

TyIsI committed:
refactor(frontend-react): moved CreateUserButton component into admin Users page component dir

TyIsI committed:
refactor(frontend-react): moved integrated pages into dedicated directory

TyIsI committed:
refactor(frontend-react): reorganized components to convention

TyIsI committed:
docs(frontend-react): improved atomic react classification section

TyIsI committed:
docs(frontend-react): further improved developer documentation

TyIsI committed:
dev(frontend-react): added create:components command and adjusted GRC destination directory

TyIsI committed:
build: refactored scripts and justfile

TyIsI committed:
chore: updated package manager

TyIsI committed:
build: update circleci config

TyIsI committed:
build: migrate tests to GHA

TyIsI committed:
build: remove pnpm version from gha run-tests workflow

TyIsI committed:
build: remove lts node version from run-tests matrix

coryalder committed:
db and backend working, react stuff still broken

coryalder committed:
Working frontend-react + frontend-web

coryalder committed:
add proxy container

coryalder committed:
nuke vagrant

coryalder committed:
clean up top navigation bar

coryalder committed:
fix sidebar styles

coryalder committed:
fix door code lookup

coryalder committed:
prevent loading spinner from hanging ui

coryalder committed:
update docker-compose script comments

coryalder committed:
make mysql use a health check instead of service started

coryalder committed:
make poxy service start up after both frontends

coryalder committed:
remove dockerfile generator

coryalder committed:
remove diff report tool and old report

coryalder committed:
move nomos.env into docker-compose directory, nuke old 'docker' directory, simplify gitignore

coryalder committed:
switch away from whitelist gitignores

coryalder committed:
remove two more gitignores

coryalder committed:
remove routeTree.gen.ts

coryalder committed:
remove ghetto deploy script and unneeded ignored files

coryalder committed:
add routeTree back

coryalder committed:
update route tree

coryalder committed:
Fix loading bug on api keys page, change spinner, add old site link to top bar

coryalder committed:
fix pagination and api key table styling

coryalder committed:
Add new 'Home' landing page

coryalder committed:
fix user granting privs

coryalder committed:
remove wireit from top-level package

coryalder committed:
remove wireit from all except frontend-react

coryalder committed:
Remove unused root dependenices x 2

coryalder committed:
add shellcheck

coryalder committed:
run shellcheck in separate job

coryalder committed:
Re-indent github actions

coryalder committed:
remove eslint config

coryalder committed:
fix webhooker tests

coryalder committed:
re-org webhooker

coryalder committed:
remove unused top level typescript configs

coryalder committed:
write about how webhooker isn't in use

coryalder committed:
nuke docker-compose composition stuff

coryalder committed:
update readme with new build instructions

coryalder committed:
Change loading spinner color

coryalder committed:
wrote about dev cycles

coryalder committed:
remove wiki link

coryalder committed:
checking readme prettier

coryalder committed:
fix readme formatting wrt prettier

coryalder committed:
formatting tweak

coryalder committed:
fromatting again

coryalder committed:
prettier md files

coryalder committed:
test prettier

coryalder committed:
delete webhooker

coryalder committed:
Nuke rabbitmq from backend-php

coryalder committed:
remove last vestiges of rabbitmq

coryalder committed:
update php version

coryalder committed:
run tests on php 8.2 instead of 8.3.16

coryalder committed:
added notes about tags / branches

acruise commented:
outfuckingstanding PR title

EricNelson12 commented on docker-compose.yml:
```diff
@@ -0,0 +1,97 @@
+            - ${PWD}/packages/frontend-web/web:/var/www/html
+
+    db-mariadb:
+        image: mariadb:11.4
```
@coryalder trying to understand all the differences in your branch. Are you moving to a single docker file instead of the docker-compose.sh switch?

And `image: mariadb:11.4` , does this switch the DB from mysql to mariadb? is that an easy migration? 

I was able to get the master branch working, but not this one so far haha

also i'm terrible at responding on github 😬

coryalder replied:
Yeah, I ended up nuking the customized docker compose script in favour of a single docker compose file. I got rid of a bunch of the extra services that were being run, so it didn't seem super useful to keep the script around. I'm VERY open to other opinions here tho, so if you think it's worth keeping we could roll that stuff back.

As I said in slack DMs, the mariadb switch was because I encountered a problem upgrading mysql - something about database adapters, that is lost to the sands of time.

coryalder committed:
symlink php, update readme to spec php 8.3 instead of 8.2

coryalder committed:
ln php in dockerfile

coryalder committed:
Auto-upgrade database

This runs mariadb's `mariadb-upgrade` tool, to migrate the data store to
the current version. Seems to need to be run twice, once to set
everything up, and then again after deleting
`./data/mysql/.my-healthcheck.cnf`
